### PR TITLE
feat: Add optional support to use compose V2

### LINKED ISSUE #272: Enhance Compose Builder to be compatible with Docker Compose V2 (cli version)
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for Compose Builder

### Description [**REQUIRED**]
 CLI version of Docker Compose is now released and being adopted. Compose Builder uses older `docker-compose` when user many not have installed. 

### Describe the solution you'd like
 Update Compose Builder to support using either `docker-compose` or `docker compose` by checking is `docker compose` is available and use it, otherwise use `docker-compose`

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by lenny-intel**
closes #272

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [x] I have opened a PR for the related docs change (if not, why?) 
  README's have been updated

## Testing Instructions
From clean system (EdgeX not running with all EdgeX containers, volumes and networks removed)
With `docker-compose` installed and `compose V2` installed
### From **top level** folder
- run `make run no-secty` 
- verify `docker-compose` was used and all services run fine
- Rename `docker-compose` so it isn't found
- run `make run no-secty` 
- verify `docker compose` was used and all services run fine
- run `make clean`
- verify `docker compose` was used
- Verify all service, containers, volumes and networks are gone.
- Restore `docker-compose` so it is found
### From **compose-builder** folder
- run `make run no-secty` 
- verify `docker-compose` with `config` command where used and all services run fine
- Rename `docker-compose` so it isn't found
- run `make run no-secty` 
- verify `docker compose` with `convert` were used and all services run fine
- run `make clean`
- verify `docker compose` was used
- Verify all service, containers, volumes and networks are gone.
- Restore `docker-compose` so it is found

### CODE REVIEW

File: compose-builder/Makefile [hunk 1]
```diff
@@ -926,14 +936,14 @@ taf-compose-perf: gen
...
+	${DOCKER_COMPOSE}  -p edgex up -d $(SERVICES)
 
 pull: gen
-	docker-compose pull $(SERVICES)
+	${DOCKER_COMPOSE}  pull $(SERVICES)
 
 gen:
 	$(GROVE_CHECK) \
-	docker-compose -p edgex $(COMPOSE_FILES) config > docker-compose.yml
+	${DOCKER_COMPOSE}  -p edgex $(COMPOSE_FILES) ${GEN_COMMAND} > docker-compose.yml
```
Comment (bnevis-i) [inline compose-builder/Makefile:946]: If we are still supporting the older docker-compose until EdgeX 3.0, shouldn't the gen command continue to use the old version?

**Review by jim-wang-intel — DISMISSED** [review]

File: compose-builder/Makefile (hunk 1 above)
Comment (lenny-intel) [reply compose-builder/Makefile:946]: It is using the version of compose you have installed. Developer (us) that run `make build` still need to use docker-compose  1.27.2 as stated is stated in the README. Use can user which ever they want.

Comment (bnevis-i) [reply compose-builder/Makefile:946]: It doesn't look that way.  Pretty much every time you install docker now, you get the compose sub-command added if you follow the docker.com instructions.  The makefile is written such that if assumes you are using docker-compose, and then if "docker compose" is found, it switches to that.  I think if you want this behavior you should assume "docker compose" is available, and then if the developer has installed a specific version of docker-compose, to use that instead.

Comment (lenny-intel) [reply compose-builder/Makefile:946]: Nope, opposite. It look for `docker-compose` installed. If found it uses it. If not is tries `docker compose`.

Comment (lenny-intel) [reply compose-builder/Makefile:946]: I have both installed. If always uses `docker-compose`. See my test instructions.

Comment (bnevis-i) [reply compose-builder/Makefile:946]: Ok.  I misread it

```
ifeq (, $(shell which docker-compose))
```

Not obvious, but I get it now.

**Review by bnevis-i — APPROVED** [review]

**Review by jim-wang-intel — DISMISSED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by jpwhitemn — APPROVED** [review]
worked for me and I have the new compose

**Pull request merged**